src/agent/src/image.rs

```diff
@@ -54,9 +54,16 @@ pub struct ImageService {
 
 impl ImageService {
     pub fn new() -> Self {
-        Self {
-            image_client: ImageClient::new(PathBuf::from(KATA_IMAGE_WORK_DIR)),
+        let mut image_client = ImageClient::new(PathBuf::from(KATA_IMAGE_WORK_DIR));
+        #[cfg(feature = "guest-pull")]
+        if !&AGENT_CONFIG.image_registry_auth.is_empty() {
```

Comment:
Am I overlooking something, or is taking a reference here not necessary?

Reply:
Yeah, I think you are correct. I'm happy to remove it